fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/tiering/source/enumerator/TieringSourceEnumerator.java

```diff
@@ -310,6 +326,55 @@ private List<TieringSplit> populateNumberOfTieringSplits(List<TieringSplit> tier
                 .collect(Collectors.toList());
     }
 
+    private List<TieringSplit> doGenerateTieringSplits(long tableId, TablePath tablePath)
+            throws Exception {
+        // Get table lake snapshot info of the given table.
+        LakeSnapshot lakeSnapshotInfo;
+        try {
+            lakeSnapshotInfo = flussAdmin.getLatestLakeSnapshot(tablePath).get();
+            LOG.info("Last committed lake table snapshot info is:{}", lakeSnapshotInfo);
+        } catch (Exception e) {
+            Throwable t = ExceptionUtils.stripExecutionException(e);
+            if (t instanceof LakeTableSnapshotNotExistException) {
+                lakeSnapshotInfo = null;
+            } else {
+                throw new FlinkRuntimeException(
+                        String.format("Failed to get table snapshot for table %s", tablePath),
+                        ExceptionUtils.stripCompletionException(e));
+            }
+        }
+
+        final TableInfo tableInfo = flussAdmin.getTableInfo(tablePath).get();
+
+        // we need to check whether the Fluss cluster is missing some lake snapshot
+        try (LakeCommitter<?, ?> lakeCommitter =
+                lakeTieringFactory.createLakeCommitter(
+                        new TieringCommitterInitContext(tablePath))) {
+            LakeCommittedSnapshot lakeCommittedSnapshot =
+                    lakeCommitter.getMissingCommittedSnapshot(
+                            lakeSnapshotInfo == null ? null : lakeSnapshotInfo.getSnapshotId());
+            if (lakeCommittedSnapshot == null) {
+                return splitGenerator.generateTableSplits(tableInfo, lakeSnapshotInfo);
+            } else {
+                // fluss still mussing offset info in lake, we commit to fluss firstly before
+                // requesting splits
+                Map<String, Long> partitionIdByName = null;
+                if (tableInfo.isPartitioned()) {
+                    partitionIdByName =
+                            flussAdmin.listPartitionInfos(tablePath).get().stream()
+                                    .collect(
+                                            Collectors.toMap(
+                                                    PartitionInfo::getPartitionName,
+                                                    PartitionInfo::getPartitionId));
+                }
+                flussTableLakeSnapshotCommitter.commit(
+                        tableId, partitionIdByName, lakeCommittedSnapshot);
+                // then, we can generate split thenAdd commentMore actions
+                return doGenerateTieringSplits(tableId, tablePath);
```

Comment:
Do we need to introduce a recursion here ？

I think we can add a method like `boolean advanceLatestLakeCommittedSnapshots()` and then call `splitGenerator.generateTableSplits` temporally . Furthermore, we'd better move this logic to committer operator.